base/distributed/remotecall.jl

```diff
@@ -16,10 +16,11 @@ mutable struct Future <: AbstractRemoteRef
     where::Int
     whence::Int
     id::Int
-    v::Nullable{Any}
+    v::Union{Some{<:Any}, Void}
 
-    Future(w::Int, rrid::RRID) = Future(w, rrid, Nullable{Any}())
-    Future(w::Int, rrid::RRID, v) = (r = new(w,rrid.whence,rrid.id,v); return test_existing_ref(r))
+    Future(w::Int, rrid::RRID) = Future(w, rrid, nothing)
+    Future(w::Int, rrid::RRID, v::Union{Some{<:Any}, Void}) =
```

Comment:
Yeah, maybe. Though I generally like that you can pass `nothing` as a second argument, this can sometimes be cleaner than calling either the two- or three-argument version depending on whether you want to set the value.

Reply:
Sure. I would just rather see it written as:
```
Future(w::Int, rrid::RRID, v::Void=nothing) = Future(w, rrid, v)
Future(w::Int, rrid::RRID, v) = Future(w, rrid, Some(v))
```
So that the code is clear about its intentions.

Is it that common to have code that maybe-sets the value?

Reply:
> Is it that common to have code that maybe-sets the value?

No idea, but I know it's hard to write generic code when you can't just pass the values to functions, but instead need to change the way the function is invoked (I suffered from this in R, which offers a `missing` function to check whether an argument was left to its default value, which isn't equivalent to explicitly passing the default value).

Reply:
Actually the two methods proposed above are problematic since that implies `Future(w, rrid, 1)` would store `Some(1)` but `Future(w, rrid, nothing)` would store `nothing` (and not `Some(nothing)`), which defeats the purpose of using `Some`. That's the same problem as allowing `convert(Some, 1)` to work. So we have to require `v::Union{Some{<:Any}, Void}`.